string  name
float64 segment_length

